### PR TITLE
Return JSON on error for JSON views.  Enhance audit logging.

### PR DESCRIPTION
**Pull request by pbugni**
When aborting from JSON views, use new `jsonify_abort()` to raise with correct headers intact.
Clean up the audit logs with consistent extra info

### CODE REVIEW

File: patientsearch/api.py [hunk 1]
```diff
@@ -80,11 +80,15 @@ def validate_auth():
...
     """Safe wrapper to lookup logged in user's identifier string for logging"""
     try:
-        user_id = oidc.user_getfield('email')
+        username = oidc.user_getfield('preferred_username')
     except Exception:
-        # mystery how token was valid at entry, but no email available?
-        user_id = "unknown"
-    return user_id
+        # mystery how token was valid at entry and now inaccessible
+        namename = "unknown"
```
Comment (ivan-c) [inline patientsearch/api.py:86]:
```suggestion
        username = "unknown"
```

File: patientsearch/api.py [hunk 2]
```diff
@@ -80,11 +80,15 @@ def validate_auth():
 def current_user_id(token):
     """Safe wrapper to lookup logged in user's identifier string for logging"""
     try:
-        user_id = oidc.user_getfield('email')
+        username = oidc.user_getfield('preferred_username')
     except Exception:
```
Comment (ivan-c) [inline patientsearch/api.py:84]: Is there a specific exception we can catch?

File: patientsearch/jsonify_abort.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+"""Utility function to bring together flask `abort` and `jsonify`"""
+from flask import abort, jsonify, make_response
+
+def jsonify_abort(**kwargs):
+    """Takes any number of args, like jsonify, but raises like abort with correct content_type heading"""
+    status_code = kwargs.pop('status_code', 500)
```
Comment (ivan-c) [inline patientsearch/jsonify_abort.py:6]: Might be nice to have `status_code` as a proper keyword arg (can be defaulted to `None`/`"500"`), so it's visible in `help(jsonify_abort)` etc

**Review by ivan-c — APPROVED** [review]
Looks good! Minor typo needs fixing, but ready to merge otherwise

File: patientsearch/api.py (hunk 1 above)
Comment (pbugni) [reply patientsearch/api.py:86]: thank you!

File: patientsearch/api.py (hunk 2 above)
Comment (pbugni) [reply patientsearch/api.py:84]: I haven't been able to reliably reproduce to catch, thus the catchall :(

File: patientsearch/jsonify_abort.py (hunk 3 above)
Comment (pbugni) [reply patientsearch/jsonify_abort.py:6]: 👍

**Pull request merged**